common/src/main/java/com/wynntils/services/wynnlanguage/WynnLanguageService.java

```diff
@@ -0,0 +1,162 @@
+    }
+
+    private void createTranslationMaps() {
+        for (int i = 0; i < gavellianCharacters.size(); i++) {
+            gavellianToEnglishMap.put(gavellianCharacters.get(i), englishCharacters.get(i));
+        }
+
+        for (int i = 0; i < wynnicCharacters.size(); i++) {
+            wynnicToEnglishMap.put(wynnicCharacters.get(i), englishCharacters.get(i));
+        }
+
+        for (int i = 0; i < gavellianCharacters.size(); i++) {
+            englishToGavellianMap.put(englishCharacters.get(i), gavellianCharacters.get(i));
+        }
+
+        for (int i = 0; i < wynnicCharacters.size(); i++) {
+            englishToWynnicMap.put(englishCharacters.get(i), wynnicCharacters.get(i));
+        }
+    }
+
+    public Character translateGavellianToEnglish(Character characterToTranslate) {
+        return gavellianToEnglishMap.getOrDefault(characterToTranslate, characterToTranslate);
+    }
+
+    public Character translateWynnicToEnglish(Character characterToTranslate) {
+        return wynnicToEnglishMap.getOrDefault(characterToTranslate, characterToTranslate);
+    }
+
+    public Character translateEnglishToGavellian(Character characterToTranslate) {
+        return englishToGavellianMap.getOrDefault(characterToTranslate, characterToTranslate);
+    }
+
+    public Character translateEnglishToWynnic(Character characterToTranslate) {
+        return englishToWynnicMap.getOrDefault(characterToTranslate, characterToTranslate);
+    }
+
+    public boolean hasTranscriber(WynnLanguage transciberToFind) {
+        Inventory inventory = McUtils.inventory();
+
+        for (int slotNum = 0; slotNum < Inventory.INVENTORY_SIZE; slotNum++) {
+            ItemStack itemStack = inventory.getItem(slotNum);
+
+            if (transciberToFind == WynnLanguage.WYNNIC) {
+                if (StyledText.fromComponent(itemStack.getHoverName()).equals(WYNNIC_TRANSCRIBER)) {
+                    return true;
+                }
+            } else {
+                if (StyledText.fromComponent(itemStack.getHoverName()).equals(GAVELLIAN_TRANSCRIBER)) {
+                    return true;
+                }
+            }
+        }
+
+        return false;
+    }
+
+    public boolean completedDiscovery(WynnLanguage discoveryToCheck) {
+        Stream<DiscoveryInfo> discoveryInfoStream =
+                Models.Discovery.getAllCompletedDiscoveries(ActivitySortOrder.ALPHABETIC);
+
+        String nameToFind = discoveryToCheck == WynnLanguage.WYNNIC
+                ? WYNNIC_TRANSCRIBER_DISCOVERY
+                : GAVELLIAN_TRANSCRIBER_DISCOVERY;
+
+        Optional<DiscoveryInfo> foundDiscoveryInfo = discoveryInfoStream
+                .filter(discoveryInfo -> discoveryInfo.getName().equals(nameToFind))
+                .findFirst();
+
+        return foundDiscoveryInfo.map(DiscoveryInfo::isDiscovered).orElse(false);
+    }
+
+    public void setSelectedLanguage(WynnLanguage selectedLanguage) {
+        this.selectedLanguage = selectedLanguage;
+    }
```

Comment:
Perhaps we should reset this after each message sent, so you don't need to set that many defaults..

Reply:
I did think that as opening the chat screen for commands via / should reset it but I couldn't figure out how to do that. This is just how legacy behaved

Reply:
Oh. Okay then, but It kind of feels messy. Maybe magicus has a better idea later.